prog/rand.go

```diff
@@ -119,10 +119,14 @@ func truncateToBitSize(v, bitSize uint64) uint64 {
 	return v & uint64(1<<bitSize-1)
 }
 
-func (r *randGen) randRangeInt(begin, end, bitSize uint64) uint64 {
+func (r *randGen) randRangeInt(begin, end, bitSize, align uint64) uint64 {
 	if r.oneOf(100) {
 		return r.randInt(bitSize)
 	}
+	if align != 0 {
+		endAlign := (end - begin) / align
```

Comment:
This needs to be (end - begin + 1).

Reply:
Do you have an example where the above doesn't work? If I do `+ 1`, `[1:100, 50]` doesn't return the expected values: `[1:100, 50] -> 1 + [0:2] * 50 -> {1, 51, 101}`.

Reply:
You are right.